prediction_market_agent/agents/logprobs_agent/deploy.py

```diff
@@ -0,0 +1,286 @@
+from concurrent.futures import ThreadPoolExecutor
+from typing import Any, Literal
+
+from prediction_market_agent_tooling.deploy.agent import DeployableTraderAgent
+from prediction_market_agent_tooling.gtypes import Probability
+from prediction_market_agent_tooling.loggers import logger
+from prediction_market_agent_tooling.logprobs_parser import LogprobsParser
+from prediction_market_agent_tooling.markets.agent_market import AgentMarket
+from prediction_market_agent_tooling.markets.data_models import ProbabilisticAnswer
+from prediction_market_agent_tooling.tools.perplexity.perplexity_models import (
+    PerplexityResponse,
+)
+from prediction_market_agent_tooling.tools.perplexity.perplexity_search import (
+    perplexity_search,
+)
+from pydantic import BaseModel, Field
+from pydantic_ai import Agent
+from pydantic_ai.agent import AgentRunResult
+from pydantic_ai.settings import ModelSettings
+
+from prediction_market_agent.agents.logprobs_oai_model import LogProbsOpenAIModel
+from prediction_market_agent.tools.openai_utils import get_openai_provider
+from prediction_market_agent.tools.web_scrape.structured_summary import (
+    web_scrape_structured_and_summarized,
+)
+from prediction_market_agent.utils import APIKeys
+
+
+class DecidabilityResponse(BaseModel):
+    rationale: str
+    answer: Literal["YES", "NO"]
+
+
+class PredictionResponse(BaseModel):
+    rationale: str
+    p_yes: float = Field(ge=0, le=1)
```

Comment:
Is there any theory behind why logprobs are used for `DecidabilityResponse`, but not for `PredictionResponse`? 🤔